File: src/web/src/types/cabinet.ts
/** 복약함 관련 타입 정의. */

export interface CabinetItem {
  id: number;
  device_id: string;
  item_type: "drug" | "supplement";
  item_id: number;
  item_name: string;
  nickname: string | null;
  created_at: string;
}

export interface CabinetItemCreate {
  item_type: "drug" | "supplement";
  item_id: number;
  nickname?: string;
}

File: src/backend/schemas/cabinet.py
"""복약함(Cabinet) 관련 API 스키마."""

from datetime import datetime

from pydantic import BaseModel, Field

from src.backend.models.user_cabinet import CabinetItemType


class CabinetItemCreate(BaseModel):
    """복약함 아이템 추가 요청."""

    item_type: CabinetItemType
    item_id: int = Field(..., gt=0)
    nickname: str | None = None


class CabinetItemResponse(BaseModel):
    """복약함 아이템 응답."""

    id: int
    item_type: CabinetItemType
    item_id: int
    item_name: str
    nickname: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}
